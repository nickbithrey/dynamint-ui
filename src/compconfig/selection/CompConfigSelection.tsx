import * as React from 'react';
import Table, { ITable, ITableColumn, ITableItem } from 'lib/table';
import Text from 'lib/text';
import { Button, IButton, ButtonContainer } from 'lib/button';
import { Key, Condition } from '~/lib';
import { Selector, ISelectionDetails, ISelectionButton, SelectionType } from 'lib/selector';

export interface ICompConfigSelection {
    id: string | number;
    loading: boolean;
    items: Array<any & ITableItem>;
    columns: Array<ITableColumn>;
    updateSelected: (selectedKeys: Array<string | number>) => void;
}

const editButton: ISelectionButton & Condition<ISelectionDetails> = {
    button: {
        key: 'edit', 
        content: 'Edit',
        type: 'primary'
    },
    linkResolver: (input: ISelectionDetails) => ({pathname: '/compconfig/details', state: { uri: input.selected[0]._uri, readOnly: false }}),
    condition: (input: ISelectionDetails) => input.selected.length > 0
};

const viewButton: ISelectionButton & Condition<ISelectionDetails> = {
    button: {
        key: 'view',
        content: 'View', 
        type: 'default'
    },
    linkResolver: (input: ISelectionDetails) => ({pathname: '/compconfig/details', state: { uri: input.selected[0]._uri, readOnly: true }}),
    condition: (input: ISelectionDetails) => input.selected.length > 0
};

const createButton: ISelectionButton & Condition<any> = {
    button: {
        key: 'create',
        content: 'Create',
        link: '/compconfigs/details',
        type: 'default'
    },
    condition: (input: ISelectionDetails) => input.selected.length === 0
};

const onItemInvoked = (item: any) => {
    const uri = item.uri;
    console.log('invoked ' + JSON.stringify(item));
}

const Selection = ({items, loading, columns, updateSelected}: ICompConfigSelection) => {
    const buttons: Array<ISelectionButton & Condition<ISelectionDetails>> = [
        viewButton,
        editButton,
        createButton
    ];
    return (
        <div>
            <Text>Configuration Selection</Text>
            <Selector
                storeKey={'config-selection'}
                loading={loading}
                items={items}
                columns={columns}
                buttons={buttons}
                selectionType={SelectionType.single}
            />
        </div>
    );
}

export default Selection;