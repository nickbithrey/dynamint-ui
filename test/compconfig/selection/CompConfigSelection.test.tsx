import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import CompConfigSelection, { ICompConfigSelection } from '~/compconfig/selection/CompConfigSelection';

describe('<CompConfigSelection />', () => {
    
    const props: ICompConfigSelection = {
        id: 'id',
        items: [
            {col1: 'val1'},
            {col1: 'val2'},
        ],
        loading: false,
        columns: [
            {key: 'col1', name: 'col1', fieldName: 'col1', minWidth: 50, maxWidth: 100}
        ],
        updateSelected: jest.fn()
    };
    
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = shallow(<CompConfigSelection {...props} />);
    });
    
    it('renders a valid tag and matches snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
});