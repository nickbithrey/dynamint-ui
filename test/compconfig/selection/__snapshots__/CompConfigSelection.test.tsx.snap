// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompConfigSelection /> renders a valid tag and matches snapshot 1`] = `
<div>
  <T>
    Configuration Selection
  </T>
  <Selector
    buttons={
      Array [
        Object {
          "button": Object {
            "content": "View",
            "key": "view",
            "type": "default",
          },
          "condition": [Function],
          "linkResolver": [Function],
        },
        Object {
          "button": Object {
            "content": "Edit",
            "key": "edit",
            "type": "primary",
          },
          "condition": [Function],
          "linkResolver": [Function],
        },
        Object {
          "button": Object {
            "content": "Create",
            "key": "create",
            "link": "/compconfigs/details",
            "type": "default",
          },
          "condition": [Function],
        },
      ]
    }
    columns={
      Array [
        Object {
          "fieldName": "col1",
          "key": "col1",
          "maxWidth": 100,
          "minWidth": 50,
          "name": "col1",
        },
      ]
    }
    items={
      Array [
        Object {
          "col1": "val1",
        },
        Object {
          "col1": "val2",
        },
      ]
    }
    loading={false}
    selectionType={1}
    storeKey="config-selection"
  />
</div>
`;
